src/anomalib/utils/visualization/image.py

```diff
@@ -139,9 +153,8 @@ def _visualize_batch(self, batch: dict) -> Iterator[GeneratorResult]:
         batch_size = batch["image"].shape[0]
         for i in range(batch_size):
             if "image_path" in batch:
-                height, width = batch["image"].shape[-2:]
-                image = (read_image(path=batch["image_path"][i]) * 255).astype(np.uint8)
-                image = cv2.resize(image, dsize=(width, height), interpolation=cv2.INTER_AREA)
+                image = batch["image"][i].cpu().numpy().transpose(1, 2, 0)  # HWC, RGB
```

Comment:
I am not sure if this is the best approach. I would recommend creating a new visualizer sub-classed from https://github.com/openvinotoolkit/anomalib/blob/56843d2671977d07ad228e6e5d870bf7f240cf59/src/anomalib/utils/visualization/base.py#L34. You can access the batch in a similar manner as https://github.com/openvinotoolkit/anomalib/blob/56843d2671977d07ad228e6e5d870bf7f240cf59/src/anomalib/utils/visualization/image.py#L124. This new class can then safely make the strong assumption that the images are always normalized to imagenet scales. Additionally, it will not break the functionality to load images from the file-path.
We are planning on refactoring the visualization callback so that we can pass the visualizers to the engine rather than hard-coding it but for now you can add the new visualizer to the list here https://github.com/openvinotoolkit/anomalib/blob/56843d2671977d07ad228e6e5d870bf7f240cf59/src/anomalib/engine/engine.py#L437

Another option is to invert the normalization transforms attached to the model https://github.com/openvinotoolkit/anomalib/blob/56843d2671977d07ad228e6e5d870bf7f240cf59/src/anomalib/models/image/padim/lightning_model.py#L131 or from the dataloader. But it might require a more complex solution.

Reply:
Thanks @ashwinvaidya17  for the feedback. I have implemented a new mechanism to invert the normalization transforms attached to the model. Please review it again.